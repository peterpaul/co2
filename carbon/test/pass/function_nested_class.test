["stdio.h"] int printf(char * format, ...);

class Test {
	int foo;

	Test () {
		foo = 1;
	}

	int test (int par) {
		int baz = 1;
		int bar (int foobar) {
			return foo + par + baz + foobar;
		}

		return bar (1);
	}
}

int main () {
	Test t = new Test ();

	printf ("t.test (1) = %d\n", t.test (1));
	printf ("t.test (2) = %d\n", t.test (2));
	printf ("t.test (3) = %d\n", t.test (3));
	printf ("t.test (4) = %d\n", t.test (4));

	return 0;
}
