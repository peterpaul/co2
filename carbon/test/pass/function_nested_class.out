t.test (1) = 4
t.test (2) = 5
t.test (3) = 6
t.test (4) = 7
